CHANGELOG.md

```diff
@@ -14,6 +14,10 @@ v0.22 + 1
   via `git_config_parse_path()` and `git_config_get_path()`
   respectively.
 
+* `git_index_add_frombuffer()` can now create a blob from memory
+   buffer and add it to the index, even if the index is also in
+   memory.
+
```

Comment:
You can create an in-memory index and associate a repository to it with `git_repository_set_index`. Maybe, I can add that the index need to be associtated to a repository.